docs/current_docs/cookbook/cookbook.mdx

```diff
@@ -193,6 +197,51 @@ The following Dagger Function accepts a `Directory` argument, which could refere
       entries
     ```
 
+### Copy a subset of a directory or remote repository to a container using pre-defined filters
+
+The following Dagger Function accepts a `Directory` argument, which could reference either a directory from the local filesystem or from a remote Git repository. It copies the specified directory to the `/src` path in a container, using pre-defined filter patterns to exclude specified sub-directories and files, and returns the modified container.
+
+:::note
+This is an example of [pre-call filtering](../api/fs-filters.mdx#pre-call-filtering) with [directory filters](../api/fs-filters.mdx).
+:::
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/copy-filter-dir-pre/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/copy-filter-dir-pre/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+```typescript file=./snippets/copy-filter-dir-pre/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+#### Examples
+
+- Copy the specified host directory to `/src` in the container, excluding everything except Markdown files, and return the modified container:
+
+    ```shell
+    dagger call copy-directory-with-exclusions --source=../docs
+    ```
+
+- Copy the `docs/current_docs` directory of the public `dagger/dagger` GitHub repository to `/src` in the container, excluding everything except Markdown files, and list the contents of the directory:
```

Comment:
@TomChv this does not seem to work. Can you help debug it?

Reply:
What's the directory content? Maybe you'll need to ignore `**/*` instead of `*`

Reply:
I wonder if it's because the source comes from a remote GH dir. It shouldn't matter though, can you also verify with a local dir to see if there's a difference?

Reply:
It works with a local dir, not with a remote dir. 

Repro:

```
admin@ip-172-31-41-139:~$ git clone https://github.com/dagger/dagger

admin@ip-172-31-41-139:~/sandbox/go$ dagger call copy-directory-with-exclusions --source=../../dagger directory --path=/src entries
✔ connect 0.2s
✔ initialize 1.0s
✔ prepare 0.1s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ ModuleSource.resolveDirectoryFromCaller(ignore: ["*", "!**/*.md"], path: "../../dagger"): Directory! 0.0s
  ): Container! 1.2s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

.changes
.dagger
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
README.md
RELEASING.md
core
dagql
docs
hack
helm
modules
sdk

admin@ip-172-31-41-139:~/sandbox/go$ dagger call copy-directory-with-exclusions --source=https://github.com/dagger/dagger#main directory --path=/src entries
✔ connect 0.2s
✔ initialize 1.1s
✔ prepare 0.0s
✔ git(keepGitDir: true, url: "https://github.com/dagger/dagger"): GitRepository! 0.0s
✔ GitRepository.ref(name: "main"): GitRef! 0.0s
✔ GitRef.tree: Directory! 3.9s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ GitRef.tree: Directory! 3.9s
  ): Container! 7.7s
✔ Container.directory(path: "/src"): Directory! 0.4s
✔ Directory.entries: [String!]! 0.0s

.changes
.changie.yaml
.dagger
.git
.github
.gitignore
.goreleaser.common.yml
.goreleaser.nightly.yml
.goreleaser.yml
.markdownlint.yaml
.readthedocs.yaml
.ruff.toml
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
NOTICE
README.md
RELEASING.md
analytics
auth
cmd
core
dagger.json
dagql
docs
engine
go.mod
go.sum
hack
helm
install.ps1
install.sh
internal
modules
network
sdk
testctx
version
yarn.lock
```

Reply:
I am seeing the same behaviour:

```console
# 👍 CONFIRMING DAGGER VERSION
❯ dagger version
dagger v0.13.7 (registry.dagger.io/engine:v0.13.7) darwin/amd64

# 👍 CONFIRMING PATH
❯ pwd
/Users/gerhard/github.com/dagger/dagger.worktree/8782/docs/current_docs/cookbook/snippets/copy-filter-dir-pre/go

# ✅ LOCAL DIR WORKS AS EXPECTED
❯ dagger call copy-directory-with-exclusions --source=../../../../../ directory --path=/src entries
✔ connect 0.3s
✔ initialize 0.8s
✔ prepare 0.1s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ ModuleSource.resolveDirectoryFromCaller(ignore: ["*", "!**/*.md"], path: "../../../../../"): Directory! 0.0s
  ): Container! 0.9s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

ARCHIVING.md
STYLE_GUIDE.md
_README.md
docs-graphql

# ❌ REMOTE DIR via GIT DOES NOT WORK AS EXPECTED
❯ dagger call copy-directory-with-exclusions --source="https://github.com/dagger/dagger#main" directory --path=/src entries
✔ connect 0.3s
✔ initialize 0.8s
✔ prepare 0.0s
✔ host: Host! 0.0s
✔ Host.unixSocket(path: "/Users/gerhard/.1password/agent.sock"): Socket! 0.0s
✔ git(
    keepGitDir: true
    sshAuthSocket: ✔ Host.unixSocket(path: "/Users/gerhard/.1password/agent.sock"): Socket! 0.0s
    url: "https://github.com/dagger/dagger"
  ): GitRepository! 0.0s
✔ GitRepository.ref(name: "main"): GitRef! 0.0s
✔ GitRef.tree: Directory! 0.3s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ GitRef.tree: Directory! 0.3s
  ): Container! 1.2s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

.changes
.changie.yaml
.dagger
.git
.github
.gitignore
.goreleaser.common.yml
.goreleaser.nightly.yml
.goreleaser.yml
.markdownlint.yaml
.readthedocs.yaml
.ruff.toml
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
NOTICE
README.md
RELEASING.md
analytics
auth
cmd
core
dagger.json
dagql
docs
engine
go.mod
go.sum
hack
helm
install.ps1
install.sh
internal
modules
network
sdk
testctx
version
yarn.lock
```

I pushed my local snippet change @vikram-dagger , but feel free to drop it if not needed.

Reply:
Thanks for re-checking this @gerhard !

Reply:
I also did a repro, seems like an established issue, I'm going to work on a fix!
We can hold this PR until the fixes is merged